dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -253,57 +244,40 @@ private static ClassSymbolReference constantToClassReference(
               + ". However, the content is not ConstantUtf8. It is "
               + classNameConstant);
     }
-    ConstantUtf8 classNameConstantUtf8 = (ConstantUtf8)classNameConstant;
+    ConstantUtf8 classNameConstantUtf8 = (ConstantUtf8) classNameConstant;
     // classNameConstantUtf8 has internal form of class names that uses '.' to separate identifiers
     String targetClassNameInternalForm = classNameConstantUtf8.getBytes();
     // Adjust the internal form to comply with binary names defined in JLS 13.1
     String targetClassName = targetClassNameInternalForm.replace('/', '.');
-
     String superClassName = sourceClass.getSuperclassName();
     boolean isSubclass = superClassName.equals(targetClassName);
 
-    ClassSymbolReference classReference =
-        ClassSymbolReference.builder()
-            .setSourceClassName(sourceClass.getClassName())
-            .setSubclass(isSubclass)
-            .setTargetClassName(targetClassName)
-            .build();
-    return classReference;
+    if (isSubclass) {
+      // A relationship between a superclass and subclass needs special validation for 'final'.
+      return new SuperClassSymbol(targetClassName);
```

Comment:
This is weird. If it's a subclass we return a SuperclassSymbol? not a SubclassSymbol?

Reply:
Updated the variable to `isInheritance`